### ISSUE TITLE
Report global object redefinition

### ISSUE THREAD

**Issue by sam-s** [pylint-dev/pylint#9807]
### Current problem

The following code redefines global objects and should be reported as such:
```
foo = 10
def f(x):
    return x + foo
foo = 100
def f(x):
    return x - foo
```

### Desired solution

It would be nice to get a warning like
```
line 4: global variable `foo` redefined (defined in line 1)
line 5: global function `f` redefined (defined in line 2)
```

### Additional context

We already have `redefined-outer-name` but it only warns when a global name is redefined _locally_, not _globally_.
